decentraland/unity-renderer
# feat: Experiences Viewer implementation

## What does this PR change?
Fixes #1707 

The current UI for portable experiences is not escalable. They still take the place of the old taskbar and grows horizontally. We need a new dedicated place for them on the left side of the screen, preferibly on a collapsable button.
Also PEX and Scenes HUDS are compeeting for the same screen space and can be colliding. We need to make sure this is in control of the players, so they decide which UIs to display. 
Portable experiences cannot be reenabled once they are disabled, and force the player to unequip and equip again. We also need to fix this.

![image](https://user-images.githubusercontent.com/64659061/156362023-ab0f4663-3080-4e4e-98ec-40e0b78ee1cc.png)

**Acceptance Criteria**
- [x] Portable experiences are all collapsed into a one single button in the HUD
- [x] Portable experiences can be disabled and REENABLED from the HUD
- [x] Portable experiences UI can be enabled/disabled individually

**Figma design**: https://www.figma.com/file/BmxUcVf2J3NhcXvlZcLHwt/PEX

## How to test the changes?
1. Using **ROPSTEN** in your Metamask plugin, go to: https://play.decentraland.zone/?NETWORK=ropsten&renderer-branch=feat/experiences-viewer
1.1. **NOTE:** If you don't have any smart wearable, use: https://play.decentraland.zone/?WITH_COLLECTIONS=43e8f43a-f260-4360-a83a-d447116cf96d&BUILDER_SERVER_URL=https://builder-api.decentraland.org/v1&NETWORK=ropsten&DEBUG_MODE=true&renderer-branch=feat/experiences-viewer
2. Equip one o several smart wearables from your backpack.
3. Notice a new "Experiences" button will appear in the taskbar.
4. Notice the related portable experiences have been loaded into the new "Experiences" HUD (the bottom left button in the taskbar).
5. From the "Experiences" HUD you will be able to show/hide the UI of each PEX or even activate/deactivate the PEX.

## Dependencies
This implementation depends on this one: https://github.com/decentraland/kernel/pull/187. Please wait for it is finished.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
